### DISCUSSION TITLE
Support for official TipTap extensions

### DISCUSSION THREAD

**Discussion by flaming-codes** [verbb/vizy#204] [Q&A] [answered]
I noticed the great enhancement with the latest release that added support for completely custom extensions. Many thanks for that great API!

But is it also possible to use official extensions listed on the TipTap site? For example, I'd be very interested in using

- [Collaboration](https://tiptap.dev/api/extensions/collaboration)
- [Collaboration Cursor](https://tiptap.dev/api/extensions/collaboration-cursor)

Not sure that is also supported, as I couldn't find any reference in the docs. Thanks!

**Comment by engram-design** [comment, marked as answer]
Full-blown collaboration isn't supported yet out of the box, because it'll likely require some extra work to integrate with Craft. You'll probably need to spin up your own Websockets process on your server - something that Vizy will seek to hand-hold you through, but nothing stopping you from DIY.

But - in theory, if you wanted to achieve _this_ level of collaboration using extensions all by yourself, you should be able to do it with Vizy's plugins, and Tiptap's API.

I've not actually _tried_ to use this extensions yet, but from a glance you should be able to include them yourself, using the [user guides](https://verbb.io/craft-plugins/vizy/user-guides/creating-a-custom-node-from-scratch) as a basis. Otherwise, happy to take a look.

Long term, we'll be looking at [hocuspocus](https://tiptap.dev/hocuspocus/) and [Tiptap Collab](https://tiptap.dev/hocuspocus/server/cloud).
